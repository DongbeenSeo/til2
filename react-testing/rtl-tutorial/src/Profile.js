import React from 'react';

const Profile = ({ usename, name }) => {
  return (
    <div>
      <b>{usename}</b>&nbsp;
      <span>({name})</span>
    </div>
  );
};

export default Profile;
