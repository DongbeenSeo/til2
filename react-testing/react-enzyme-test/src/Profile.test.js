import React from 'react';
import { mount } from 'enzyme';
import Profile from './Profile';

describe('<Profile/>', () => {
  it('matches snapshot', () => {
    const wrapper = mount(<Profile username="dongbeen" name="서동빈" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders usename and name', () => {
    const wrapper = mount(<Profile username="dongbeen" name="서동빈" />);
    console.log(`${JSON.stringify(wrapper.props())}`);
    expect(wrapper.props().username).toBe('dongbeen');
    expect(wrapper.props().name).toBe('서동빈');

    const boldElement = wrapper.find('b');
    expect(boldElement.contains('dongbeen')).toBe(true);
    const spanElement = wrapper.find('span');
    expect(spanElement.text()).toBe('(서동빈)');
  });
});
