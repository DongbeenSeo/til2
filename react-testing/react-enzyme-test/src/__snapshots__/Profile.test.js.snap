// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile/> matches snapshot 1`] = `
<Profile
  name="서동빈"
  username="dongbeen"
>
  <div>
    <b>
      dongbeen
    </b>
     
    <span>
      (
      서동빈
      )
    </span>
  </div>
</Profile>
`;
